cli/run.go

```diff
@@ -222,6 +222,10 @@ func (r WorkflowRunner) runWorkflows(tracker analytics.Tracker) (models.BuildRun
 
 	buildIDProperties := coreanalytics.Properties{analytics.BuildExecutionID: uuid.Must(uuid.NewV4()).String()}
 
+	if err := configs.RegisterAgentOverrides(); err != nil {
+		return models.BuildRunResultsModel{}, fmt.Errorf("apply agent config: %s", err)
+	}
```

Comment:
Now this checks happens in the workflow execution process but to me this seems like not directly linked to it. If we can then I would put this somewhere earlier in the chain. Like as high level as we can put it. Maybe before this line:
https://github.com/bitrise-io/bitrise/blob/cbfa1bb2a14a08318f980bfd2bfacb380f25be8b/cli/run.go#L88

Reply:
Interesting, my mental model is that all of these agent-specific modifications are only relevant to the `bitrise run` subcommand. Do you think it should apply to other subcommands too?

Reply:
Yes, I agree with you that this is `bitrise run` specific. The code I linked is in the `run.go` file which is hosting the `bitrise run` subcommand:
https://github.com/bitrise-io/bitrise/blob/cbfa1bb2a14a08318f980bfd2bfacb380f25be8b/cli/run.go#L50-L71

What I was referring to is that where this logic is now is a place where the actual execution flow is calculated inside the `WorkflowRunner` struct. This agent dir override seemed like something this runner should not be even aware and that is why I thought moving it out to the very beginning of the `run` subcommand execution could make sense.